File: fluttter/simple_app/lib/domain/models/lecture.dart
import 'package:simple_app/domain/models/course.dart';
import 'package:simple_app/domain/models/professor.dart';
import 'package:simple_app/domain/models/room.dart';

import 'enums/day.dart';

class Lecture {
  int id;
  String? name;
  int day;
  double timeFrom;
  double timeTo;
  Professor professor;
  Course course;
  Room room;

  Lecture({
    required this.id,
    required this.name,
    required this.day,
    required this.timeFrom,
    required this.timeTo,
    required this.professor,
    required this.course,
    required this.room,
  });

  factory Lecture.fromJson(Map<String, dynamic> json) {
    return Lecture(
        id: json['id'],
        name: json['name'],
        day: json['day'],
        timeFrom: double.parse(json['timeFrom'].substring(0, 2)),
        timeTo: double.parse(json['timeTo'].substring(0, 2))+1,
        professor: Professor.fromJson(json['professor']),
        course: Course.fromJson(json['course']),
        room: Room.fromJson(json['room']));
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'day': day,
      'timeFrom': timeFrom,
      'timeTo': timeTo,
      'professorId': professor.toJson(),
      'courseId': course.toJson(),
      'roomName': room.toJson(),
    };
  }
}

File: dotnet/FinkiRasporedi/Models/Domain/Lecture.cs
using FinkiRasporedi.Models.Enums;
using System.ComponentModel.DataAnnotations;

namespace FinkiRasporedi.Models.Domain
{
    public class Lecture
    {
        [Key]
        public int Id { get; set; }
        public string? Name { get; set; }
        public Day Day { get; set; }
        public int TimeFrom { get; set; }
        public int TimeTo { get; set; }
        
        public virtual Professor Professor { get; set; }
        public virtual Course Course { get; set; }
        public virtual Room Room { get; set; }
    }
}
